python/python/bystro/ancestry/tests/test_ancestry_types.py

```diff
@@ -1,16 +1,13 @@
 """Test ancestry_types.py."""
+import re
```

Comment:
Most of these tests are not needed. The entire point using something like msgspec is to rely on the static type system that python offers, and which is enforced by mypy, at compile time. Previously, this was obvious: we had to add # type: ignore assertions everywhere to satisfy mypy, because it caught those errors.

So I've elected to keep test_SuperpopVector_rejects_missing_key and test_SuperpopVector_rejects_extra_key to prove that even if mypy didn't catch the errors, msgspec strict mode prevents unknown parameters from being passed in, like expected, and then only tested the unit validator, which cannot be enforced by mypy currently (there is a solution I'm exploring, but its outside the scope of this PR).